sound/soc/sof/pcm.c

```diff
@@ -473,6 +473,12 @@ static int sof_pcm_trigger(struct snd_soc_component *component,
 		/* free widget list only for SUSPEND trigger */
 		if (free_widget_list)
 			ret = sof_widget_list_free(sdev, spcm, substream->stream);
+
+		ret = snd_sof_pcm_platform_hw_free(sdev, substream);
+		if (ret < 0) {
+			dev_err(sdev->dev, "platform_hw_free failed during trigger cmd %d\n", cmd);
+			return ret;
+		}
```

Comment:
this is too HDaudio-specific. There's no reason in general why we should conflate stop and hw_free.

I am already surprised we reset the hw_params in stop, that's not the intended behavior in general.

that should be in a different patch anyways.

Reply:
Should it be moved under `f (!ret && reset_hw_params)` few lines up?

Reply:
The `reset_hw_params` is only set in case of `SNDRV_PCM_TRIGGER_STOP`

Reply:
no it is set for both stop and suspend.